### PR TITLE
apis: add GatewayClassConditionReason Unsupported

### PR DESCRIPTION
**Pull request by mikemorris**
...and clarifies the description for when GatewayClassConditionReason InvalidParameters should be preferred.

I considered adding something like `GatewayClassConditionType ResolvedRefs` with `GatewayClassConditionType InvalidParametersRef` as a more spec-consistent way to handle an invalid `parametersRef` _reference_ as distinct from malformed content, but that felt like too big a change given that the documentation for the [`parametersRef`](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.GatewayClassSpec) field currently states:

> If the referent cannot be found, the GatewayClass’s “InvalidParameters” status condition will be true.

**EDIT:** Although now that I read that, that sentence needs to be reworded to be accurate.

**What type of PR is this?**
/kind documentation
/area conformance

**What this PR does / why we need it**:
Adds an appropriate reason for implementations (such as those of cloud vendors) which provide their own fixed GatewayClass offerings and do not support manually created GatewayClass resources.

**Which issue(s) this PR fixes**:
Provides an appropriate status condition reason to set for the `GatewayClassObservedGeneration` test currently skipped and blocking Azure Application Gateway for Containers full conformance in https://github.com/kubernetes-sigs/gateway-api/pull/3021

**Does this PR introduce a user-facing change?**:
```release-note
Adds the GatewayClassConditionReason "Unsupported" and clarifies when GatewayClassConditionReason "InvalidParameters" should be preferred.
```

/cc @robscott @youngnick @snehachhabria

### CODE REVIEW

**Review by robscott — COMMENTED** [review]
Thanks @mikemorris!

File: apis/v1/gatewayclass_types.go [hunk 1]
```diff
@@ -93,8 +92,10 @@ type GatewayClassSpec struct {
 	// or an implementation-specific custom resource. The resource can be
 	// cluster-scoped or namespace-scoped.
 	//
-	// If the referent cannot be found, the GatewayClass's "InvalidParameters"
-	// status condition will be true.
+	// If the referent cannot be found, refers to an unsupported kind, or when
+	// the data within that resource is malformed, the GatewayClass will be
```
Comment (mikemorris) [inline apis/v1/gatewayclass_types.go:96]:
```suggestion
	// the data within that resource is malformed, the GatewayClass may be
```
I don't think we enforce this behavior with any conformance tests currently, and I'm curious what some implementations that don't follow the "snapshotting" behavior do here - flipping a GatewayClass to `{ type: Accepted, status: False }` by setting a bad parameter could be pretty destructive to any Gateways with a "live" reference.

File: apis/v1/gatewayclass_types.go [hunk 2]
```diff
@@ -177,8 +178,13 @@ const (
 	GatewayClassReasonAccepted GatewayClassConditionReason = "Accepted"
 
 	// This reason is used with the "Accepted" condition when the
-	// GatewayClass was not accepted because the parametersRef field
-	// was invalid, with more detail in the message.
+	// GatewayClass was not accepted, with more detail in the message.
+	GatewayClassReasonInvalid GatewayClassConditionReason = "Invalid"
```
[the remark this answers is not in the document]
Comment (mikemorris) [reply apis/v1/gatewayclass_types.go:183]: I'd be happy with `Unsupported` if folks would prefer that!

File: apis/v1/gatewayclass_types.go (hunk 1 above)
Comment (robscott) [reply apis/v1/gatewayclass_types.go:97]: Good catch - in general we try to use RFC language in the spec. I'd recommend using "SHOULD" in this context.

File: apis/v1/gatewayclass_types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (snehachhabria) [reply apis/v1/gatewayclass_types.go:184]: I feel `Unsupported` is a better than `Invalid` as its a more accurate region. Invalid can have a wider scope

File: apis/v1/gatewayclass_types.go (hunk 1 above)
Comment (mikemorris) [reply apis/v1/gatewayclass_types.go:97]:
```suggestion
	// the data within that resource is malformed, the GatewayClass SHOULD be
```

**Comment by robscott** [comment]
@mikemorris would it be ok if we held off on merging this until post-v1.1? I'm completely on board with the change, but hate to make any changes to CRDs/API Spec between RC2 and final release. Maybe implementations could just manually add this condition until it was formally added here?

**Comment by robscott** [comment]
>Is this Docker builder removal timeout issue with CI documented anywhere as needing a fix yet?

Nope, I think I asked about in Slack once, but really unsure who could help us figure this one out. As a short term fix, it might be worth splitting out our docker build verification into a separate task that doesn't run with the rest of our verification. We'd just need to rework our Makefile a bit and add a new task here: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/gateway-api/gateway-api-config.yaml.

**Comment by mikemorris** [comment]
> @mikemorris would it be ok if we held off on merging this until post-v1.1?

Sure, no objections to that!

**Comment by robscott** [comment]
Thanks @mikemorris!

/lgtm
/approve

**Pull request merged**